src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const arrayCopy = [...this];
+    const len = this.length;
+
+    let newSeparator = '' + separator;
+    let result = '';
+
+    for (let i = 0; i < len; i++) {
+      if (i === len - 1) {
+        newSeparator = '';
+      }
+
+      if (this[i] === null) {
+        arrayCopy[i] = '';
+      }
+
+      if (this[i] === undefined) {
+        arrayCopy[i] = '';
+      }
```

Comment:
t would be better to combine this into one if statement using the && operator

Reply:
@DidenkoDenis306 

yeah for lines 19 - 25, it can be, but with the || operator. I'll do it